packages/web/src/pages/sign-up-page/components/NavHeader.tsx

```diff
@@ -0,0 +1,115 @@
+import { useCallback } from 'react'
+
+import {
+  Box,
+  Flex,
+  IconAudiusLogoHorizontal,
+  IconCaretLeft,
+  IconCloseAlt,
+  PlainButton,
+  PlainButtonSize,
+  PlainButtonType,
+  iconSizes
+} from '@audius/harmony'
+import { Route, Switch, useHistory, useRouteMatch } from 'react-router-dom'
+
+import { getSignOn } from 'common/store/pages/signon/selectors'
+import { useMedia } from 'hooks/useMedia'
+import { useSelector } from 'utils/reducer'
+import {
+  SIGN_UP_ARTISTS_PAGE,
+  SIGN_UP_EMAIL_PAGE,
+  SIGN_UP_FINISH_PROFILE_PAGE,
+  SIGN_UP_GENRES_PAGE,
+  SIGN_UP_HANDLE_PAGE,
+  TRENDING_PAGE
+} from 'utils/route'
+
+import { determineAllowedRoute } from '../utils/determineAllowedRoutes'
+
+import { ProgressHeader } from './ProgressHeader'
+
+export const useIsBackAllowed = () => {
+  const match = useRouteMatch<{ currentPath: string }>('/signup/:currentPath')
+  const signUpState = useSelector(getSignOn)
+  if (match?.params.currentPath) {
+    const { allowedRoutes } = determineAllowedRoute(
+      signUpState,
+      match?.params.currentPath
+    )
```

Comment:
optionally, you could make `determineAllowedRoute` a hook as well and it can call `useSelector` inside

Reply:
i was so down with that, but then i saw the other reference to it was in a "render" function and i think? that cant use a hook